### PR TITLE
Refactor CardView

### PR DESCRIPTION
**Pull request by caiofaustino**
## Summary
- Refactor CardView to Kotlin
- Use ViewBinding

### CODE REVIEW

File: card/src/main/java/com/adyen/checkout/card/CardView.kt [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+        binding.switchStorePaymentMethod.setOnCheckedChangeListener { _, isChecked ->
+            mCardInputData.isStorePaymentSelected = isChecked
+            notifyInputDataChanged()
+        }
+        if (component.isStoredPaymentMethod()) {
+            component.getStoredPaymentInputData()?.let {
+                setStoredCardInterface(it)
+            }
+        } else {
+            binding.textInputLayoutCardHolder.visibility = if (component.isHolderNameRequire()) VISIBLE else GONE
```
Comment (jreij) [inline card/src/main/java/com/adyen/checkout/card/CardView.kt:79]: What do you think if we add an extension function like this:

```kotlin
fun View.setVisible(visible: Boolean) {
    visibility = if (visible) View.VISIBLE else View.GONE
}
```

This way we can just use `binding.textInputLayoutCardHolder.setVisible(component.isHolderNameRequire()))`

If you like the idea I can open a new PR and do a refactor for the whole project in it.

File: card/src/main/java/com/adyen/checkout/card/CardView.kt [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+        }
+        if (detectedCardTypes.isEmpty()) {
+            binding.cardBrandLogoImageView.setStrokeWidth(0f)
+            binding.cardBrandLogoImageView.setImageResource(R.drawable.ic_card)
+            binding.editTextCardNumber.setAmexCardFormat(false)
+        } else {
+            binding.cardBrandLogoImageView.setStrokeWidth(RoundCornerImageView.DEFAULT_STROKE_WIDTH)
+            mImageLoader?.load(detectedCardTypes[0].cardType.txVariant, binding.cardBrandLogoImageView)
+            // TODO: 29/01/2021 get this logic from OutputData
+            var isAmex = false
```
Comment (jreij) [inline card/src/main/java/com/adyen/checkout/card/CardView.kt:191]: You can use `val isAmex = detectedCardTypes.any { it.cardType == CardType.AMERICAN_EXPRESS }`

File: card/src/main/java/com/adyen/checkout/card/CardView.kt (hunk 1 above)
Comment (caiofaustino) [reply card/src/main/java/com/adyen/checkout/card/CardView.kt:79]: What about this?
```kotlin
var View.visible: Boolean
    get() = this.visibility == VISIBLE
    set(value) {
        if (value) this.visibility = VISIBLE
        else this.visibility = GONE
    }
```

Comment (jreij) [reply card/src/main/java/com/adyen/checkout/card/CardView.kt:79]: Even better! Btw you can remove `this` (I think)

Comment (caiofaustino) [reply card/src/main/java/com/adyen/checkout/card/CardView.kt:79]: Yeah, OK, I'll add this, I was thinking if we wanted to make the view `INVISIVLE` instead, but those are less likely and we can always just set it manually.

**Review by jreij — DISMISSED** [review]

**Review by jreij — APPROVED** [review]

**Pull request merged**